### DISCUSSION TITLE
error with sdf when using SDMolSupplier

### DISCUSSION THREAD

**Discussion by migueltarancon** [rdkit/rdkit#7337] [Q&A] [answered]
Hi all!

Recently, I have encountered an error with a molecule in SDF format when trying to process it with a SDMolSupplier. The molecule has some aromatic bonds, when I replace them with single and double bonds, it works fine. The snippet of code where I have the error is:

```
from rdkit import Chem

suppl = Chem.rdmolfiles.SDMolSupplier('mol_ok.sdf')
#suppl = Chem.rdmolfiles.SDMolSupplier('mol_fail.sdf')

mol = next(suppl)
print(mol.GetNumAtoms())
```

If I use mol_ok.sdf (without aromatic bonds) everything works just fine, but when I use mol_fail.sdf (with aromatic bonds) it crashes. It gives me this error: `ERROR: Can't kekulize mol.  Unkekulized atoms: 25 26 27 28 29 32 33 34 35`, so I suppose there is a problem with the molecule. Can anyone explain what is happening?

I adjoint both SDF files. (As I can't add files with .sdf extension I have converted them to .txt files)

[mol_fail.txt](https://github.com/rdkit/rdkit/files/14886696/mol_fail.txt)
[mol_ok.txt](https://github.com/rdkit/rdkit/files/14886697/mol_ok.txt)

**Comment by greglandrum** [comment, marked as answer]
@migueltarancon : According to the documentation, SDF files are only supposed to include aromatic bond types if they are encoding queries. It's better to provide the structures with explicit single and double bonds and thus both follow the format spec and avoid all problems with incompatible aromaticity models.

**Reply by migueltarancon** [reply]
Thanks for your reply, Greg!
